In [1]:
import pandas as pd

# Loading the datasets
purch_data = pd.read_csv('purchase_behaviour.csv')
trans_data = pd.read_csv('transaction_data.csv')

### Purchase Data Analysis

In [2]:
purch_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 72637 entries, 0 to 72636
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   LYLTY_CARD_NBR    72637 non-null  int64 
 1   LIFESTAGE         72637 non-null  object
 2   PREMIUM_CUSTOMER  72637 non-null  object
dtypes: int64(1), object(2)
memory usage: 1.7+ MB


In [3]:
print("\nMissing Values:")
print(purch_data.isnull().sum())


Missing Values:
LYLTY_CARD_NBR      0
LIFESTAGE           0
PREMIUM_CUSTOMER    0
dtype: int64


In [4]:
print("Unique values in 'LIFESTAGE':")
print(purch_data['LIFESTAGE'].unique())

Unique values in 'LIFESTAGE':
['YOUNG SINGLES/COUPLES' 'YOUNG FAMILIES' 'OLDER SINGLES/COUPLES'
 'MIDAGE SINGLES/COUPLES' 'NEW FAMILIES' 'OLDER FAMILIES' 'RETIREES']


In [5]:
print("\nDistribution of LIFESTAGE:")
print(purch_data['LIFESTAGE'].value_counts())


Distribution of LIFESTAGE:
RETIREES                  14805
OLDER SINGLES/COUPLES     14609
YOUNG SINGLES/COUPLES     14441
OLDER FAMILIES             9780
YOUNG FAMILIES             9178
MIDAGE SINGLES/COUPLES     7275
NEW FAMILIES               2549
Name: LIFESTAGE, dtype: int64


In [6]:
print("\nUnique values in 'PREMIUM_CUSTOMER':")
print(purch_data['PREMIUM_CUSTOMER'].unique())


Unique values in 'PREMIUM_CUSTOMER':
['Premium' 'Mainstream' 'Budget']


In [7]:
print("\nDistribution of PREMIUM_CUSTOMER:")
print(purch_data['PREMIUM_CUSTOMER'].value_counts())


Distribution of PREMIUM_CUSTOMER:
Mainstream    29245
Budget        24470
Premium       18922
Name: PREMIUM_CUSTOMER, dtype: int64


### Transaction Data anlaysis

In [8]:
print(trans_data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 264836 entries, 0 to 264835
Data columns (total 8 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   DATE            264836 non-null  int64  
 1   STORE_NBR       264836 non-null  int64  
 2   LYLTY_CARD_NBR  264836 non-null  int64  
 3   TXN_ID          264836 non-null  int64  
 4   PROD_NBR        264836 non-null  int64  
 5   PROD_NAME       264836 non-null  object 
 6   PROD_QTY        264836 non-null  int64  
 7   TOT_SALES       264836 non-null  float64
dtypes: float64(1), int64(6), object(1)
memory usage: 16.2+ MB
None


In [9]:
print("\nMissing Values:")
print(trans_data.isnull().sum())


Missing Values:
DATE              0
STORE_NBR         0
LYLTY_CARD_NBR    0
TXN_ID            0
PROD_NBR          0
PROD_NAME         0
PROD_QTY          0
TOT_SALES         0
dtype: int64


In [10]:
print("\n Statistics of Transaction Data:")
print(trans_data.describe())


 Statistics of Transaction Data:
                DATE     STORE_NBR  LYLTY_CARD_NBR        TXN_ID  \
count  264836.000000  264836.00000    2.648360e+05  2.648360e+05   
mean    43464.036260     135.08011    1.355495e+05  1.351583e+05   
std       105.389282      76.78418    8.057998e+04  7.813303e+04   
min     43282.000000       1.00000    1.000000e+03  1.000000e+00   
25%     43373.000000      70.00000    7.002100e+04  6.760150e+04   
50%     43464.000000     130.00000    1.303575e+05  1.351375e+05   
75%     43555.000000     203.00000    2.030942e+05  2.027012e+05   
max     43646.000000     272.00000    2.373711e+06  2.415841e+06   

            PROD_NBR       PROD_QTY      TOT_SALES  
count  264836.000000  264836.000000  264836.000000  
mean       56.583157       1.907309       7.304200  
std        32.826638       0.643654       3.083226  
min         1.000000       1.000000       1.500000  
25%        28.000000       2.000000       5.400000  
50%        56.000000       2.000000

In [11]:
print("\nSales Distribution:")
print(trans_data['TOT_SALES'].describe())


Sales Distribution:
count    264836.000000
mean          7.304200
std           3.083226
min           1.500000
25%           5.400000
50%           7.400000
75%           9.200000
max         650.000000
Name: TOT_SALES, dtype: float64


In [12]:
print("\n Total unique values in 'PROD_NAME':")
print(trans_data['PROD_NAME'].nunique())


 Total unique values in 'PROD_NAME':
114


Finding top-selling products

In [13]:
top_products = trans_data.groupby('PROD_NAME')['TOT_SALES'].sum().reset_index()
top_products_sorted = top_products.sort_values(by='TOT_SALES', ascending=False).head(10)

print("\nTop 10 Bestselling Products:")
print(top_products_sorted[['PROD_NAME', 'TOT_SALES']])


Top 10 Bestselling Products:
                                   PROD_NAME  TOT_SALES
11          Dorito Corn Chp     Supreme 380g    40352.0
86    Smiths Crnkle Chip  Orgnl Big Bag 380g    36367.6
77  Smiths Crinkle Chips Salt & Vinegar 330g    34804.2
33    Kettle Mozzarella   Basil & Pesto 175g    34457.4
76         Smiths Crinkle      Original 330g    34302.6
6                       Cheezels Cheese 330g    34296.9
12          Doritos Cheese      Supreme 330g    33390.6
39   Kettle Sweet Chilli And Sour Cream 175g    33031.8
34                      Kettle Original 175g    32740.2
35      Kettle Sea Salt     And Vinegar 175g    32589.0


**Merging the datasets based on 'LYLTY_CARD_NBR**

In [14]:
merged_data = pd.merge(trans_data, purch_data, on='LYLTY_CARD_NBR', how='inner')

merged_data.head()

DATE  STORE_NBR  LYLTY_CARD_NBR  TXN_ID  PROD_NBR  \
0  43390          1            1000       1         5   
1  43599          1            1307     348        66   
2  43414          1            1307     346        96   
3  43533          1            1307     347        54   
4  43605          1            1343     383        61   

                                PROD_NAME  PROD_QTY  TOT_SALES  \
0  Natural Chip        Compny SeaSalt175g         2        6.0   
1                CCs Nacho Cheese    175g         3        6.3   
2          WW Original Stacked Chips 160g         2        3.8   
3                       CCs Original 175g         1        2.1   
4  Smiths Crinkle Cut  Chips Chicken 170g         2        2.9   

                LIFESTAGE PREMIUM_CUSTOMER  
0   YOUNG SINGLES/COUPLES          Premium  
1  MIDAGE SINGLES/COUPLES           Budget  
2  MIDAGE SINGLES/COUPLES           Budget  
3  MIDAGE SINGLES/COUPLES           Budget  
4  MIDAGE SINGLES/COUPLES           Budget

### Displaying top 3 most profitable products

In [34]:
top_products = merged_data.groupby('PROD_NAME')['TOT_SALES'].sum().reset_index().sort_values(by='TOT_SALES', ascending=False)

print("Top 3 Most Profitable Products:")
print(top_products.head(3))

Top 3 Most Profitable Products:
                                   PROD_NAME  TOT_SALES
11          Dorito Corn Chp     Supreme 380g    40352.0
86    Smiths Crnkle Chip  Orgnl Big Bag 380g    36367.6
77  Smiths Crinkle Chips Salt & Vinegar 330g    34804.2


**Displaying top PREMIUM_CUSTOMER and Lifestage based on total sales**

In [18]:
# Group by 'LIFESTAGE' and calculate total sales for each group
loyal_customers_by_lifestage = merged_data.groupby('LIFESTAGE')['TOT_SALES'].sum().reset_index().sort_values(by='TOT_SALES', ascending=False)

# Group by 'PREMIUM_CUSTOMER' and calculate total sales for each group
loyal_customers_by_premium = merged_data.groupby('PREMIUM_CUSTOMER')['TOT_SALES'].sum().reset_index().sort_values(by='TOT_SALES', ascending=False)

# Display results
print("\nTotal Sales by LIFESTAGE:")
print(loyal_customers_by_lifestage)

print("\nTotal Sales by PREMIUM_CUSTOMER:")
print(loyal_customers_by_premium)



Total Sales by LIFESTAGE:
                LIFESTAGE  TOT_SALES
3   OLDER SINGLES/COUPLES  402426.75
4                RETIREES  366470.90
2          OLDER FAMILIES  353767.20
5          YOUNG FAMILIES  316160.10
6   YOUNG SINGLES/COUPLES  260405.30
0  MIDAGE SINGLES/COUPLES  184751.30
1            NEW FAMILIES   50433.45

Total Sales by PREMIUM_CUSTOMER:
  PREMIUM_CUSTOMER  TOT_SALES
1       Mainstream  750744.50
0           Budget  676211.55
2          Premium  507458.95


## Key FIndings

**1. Spending based on Lifestage**
Older customers (Older Singles/Couples, Retirees, Older Families) spend the most.This could be because they have more disposable income, prefer convenience, or choose health-related products.

Young Families and Young Singles/Couples also spend a lot, but less than older groups.

New Families spend the least, possibly because they are still building their shopping habits or buy fewer expensive items.

**2. Spending based on Customer type**
Most sales come from Mainstream customers, who likely prefer affordable, good-quality products.

Budget customers are also a big group, probably looking for lower prices.

Premium customers spend less overall, even though they buy more expensive products. They might be more selective and buy less often.

### Product Segmentation by Customer Segment:
To analyze which products are most popular among specific customer segments, we'll group the merged dataset by LIFESTAGE and PREMIUM_CUSTOMER and calculate the total sales of each product within these segments

In [19]:
products_by_lifestage = merged_data.groupby(['LIFESTAGE', 'PROD_NAME'])['TOT_SALES'].sum().reset_index()

products_by_lifestage_sorted = products_by_lifestage.sort_values(['LIFESTAGE', 'TOT_SALES'], ascending=[True, False])

print("Products by Lifestage:")
print(products_by_lifestage_sorted.head(10))

products_by_premium_customer = merged_data.groupby(['PREMIUM_CUSTOMER', 'PROD_NAME'])['TOT_SALES'].sum().reset_index()

products_by_premium_sorted = products_by_premium_customer.sort_values(['PREMIUM_CUSTOMER', 'TOT_SALES'], ascending=[True, False])

print("\nProducts by Premium Customer:")
print(products_by_premium_sorted.head(10))


Products by Lifestage:
                 LIFESTAGE                                 PROD_NAME  \
86  MIDAGE SINGLES/COUPLES    Smiths Crnkle Chip  Orgnl Big Bag 380g   
6   MIDAGE SINGLES/COUPLES                      Cheezels Cheese 330g   
11  MIDAGE SINGLES/COUPLES          Dorito Corn Chp     Supreme 380g   
39  MIDAGE SINGLES/COUPLES   Kettle Sweet Chilli And Sour Cream 175g   
33  MIDAGE SINGLES/COUPLES    Kettle Mozzarella   Basil & Pesto 175g   
77  MIDAGE SINGLES/COUPLES  Smiths Crinkle Chips Salt & Vinegar 330g   
35  MIDAGE SINGLES/COUPLES      Kettle Sea Salt     And Vinegar 175g   
34  MIDAGE SINGLES/COUPLES                      Kettle Original 175g   
76  MIDAGE SINGLES/COUPLES         Smiths Crinkle      Original 330g   
31  MIDAGE SINGLES/COUPLES                        Kettle Chilli 175g   

    TOT_SALES  
86     3746.5  
6      3648.0  
11     3646.5  
39     3418.2  
33     3385.8  
77     3351.6  
35     3321.0  
34     3250.8  
76     3249.0  
31     3202.2  

Product

## Key Findings from the Data

**1.Most Popular Products by Lifestage (Midage Singles/Couples):**
The top-selling products for Midage Singles/Couples are mostly snack chips.
Smiths Crinkle Chips, Kettle Chips, and Doritos are among the most popular.

Larger pack sizes (e.g., 330g, 380g) are preferred, suggesting they may buy in bulk or for group consumption.

**2. Most Popular Products by Premium Customer Type (Budget Customers):**
Budget customers mainly purchase large-pack chips like Doritos, Smiths Crinkle, and Kettle chips.
This group spends the most on snacks despite being price-conscious.
They prefer value-for-money products, possibly buying in bulk or choosing larger packs to save money.

**Business Takeaways:**
Snack brands like Smiths, Kettle, and Doritos dominate sales across customer groups.

Larger pack sizes sell better, indicating bulk buying or preference for value-based purchases.

### Customer Retention:
To identify repeat customers and analyze their frequency of purchases, we can check how many times each customer (LYLTY_CARD_NBR) appears in the transaction dataset. This will help us understand the loyalty of customers.

In [20]:
customer_transactions = merged_data.groupby('LYLTY_CARD_NBR').size().reset_index(name='TXN_COUNT')

repeat_customers = customer_transactions[customer_transactions['TXN_COUNT'] > 1]
repeat_customers_sorted = repeat_customers.sort_values(by='TXN_COUNT', ascending=False)

print("\nTop Repeat Customers:")
print(repeat_customers_sorted.head(10))



Top Repeat Customers:
       LYLTY_CARD_NBR  TXN_COUNT
42813          162039         18
45905          172032         18
34943          128178         17
60925          230078         17
31654          116181         17
3686            13138         17
25958           94185         16
6530            23192         16
28605          105026         16
30744          113080         16


## Key Findings on Repeat Customers

**Most Frequent Shoppers:**
The top two customers (162039 & 172032) made 18 purchases each.

Several other customers made 16-17 purchases, showing strong repeat buying behavior.